### ISSUE TITLE
Deprecate `_type` and use `type_` instead

### ISSUE THREAD

**Issue by praveenperera** [ahrefs/bs-reactstrap#19]
Should we make this change to keep it consistent with ReasonReact?

**Comment by liubko** [comment]
this is how it works now

https://reasonml.github.io/en/try.html?rrjsx=true&reason=NoAQRgzgdA9mBWBdAUAUwB4BdUCcB2AhgDYAEAtgQNaoAKOMADhCQFwkAUyJJAfgPqYAng1RsImHAEs8AcwC8AfgA0XXhEkAvUSXFTZildwCueSZmQBKEnIB8JPtZIAiJwG5kyIqkwkCjitR0jBDs-EIick7Y4k5KHBYWrkA

we have to use `_type` anyway, so that the result object would have `type` key


----

what we can do is to accept a `type_` in `make` function, and then pass it as `_type` to `makeProps ` function, this will allow us to have the api consistent with ReasonReact

```
[@bs.obj]
external makeProps:
  (
    ~_type: string=?,
    ~size: string=?,
    unit
  ) =>
  _ =
  "";

let make =
    (
      ~type_=?,
      ~size=?,
      children,
    ) =>
  ReasonReact.wrapJsForReason(
    ...
      makeProps(
        ~_type=?type_,
        ~size?,
        (),
      ),
    ...
  );
```

**Comment by praveenperera** [comment]
I like this idea. 

I can submit a PR to make the changes. But this will break existing code. 

Does it make sense to accept both for the time being with a deprecation message? Is it possible to show a depreciation message?

**Comment by liubko** [comment]
@rusty-key @feihong what do you think?

----

_I'm ok with breaking changes_

**Comment by rusty-key** [comment]
I am ok with breaking changes too, we are still at "experimental" phase.

**Comment by liubko** [comment]
published `v0.5.3`, @praveenperera thanks for contribution!
